apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -25,17 +33,27 @@ const getById = (id: number) =>
   db.selectFrom('Resource').where('Resource.id', '=', id)
 
 // NOTE: Throw here to fail early if our invariant that a page has a `blobId` is violated
-export const getFullPageById = (id: number) => {
+export const getFullPageById = async (id: number) => {
+  // Check if draft blob exists and return that preferentially
+  const draftBlob = await getById(id)
+    .where('Resource.draftBlobId', 'is not', null)
+    .innerJoin('Blob', 'Resource.draftBlobId', 'Blob.id')
+    .selectAll()
+    .executeTakeFirst()
+  if (draftBlob) return draftBlob
+
   return getById(id)
-    .where('Resource.blobId', 'is not', null)
-    .innerJoin('Blob', 'Resource.blobId', 'Blob.id')
+    .where('Resource.publishedBlobId', 'is not', null)
+    .innerJoin('Blob', 'Resource.publishedBlobId', 'Blob.id')
     .selectAll()
     .executeTakeFirstOrThrow()
 }
```

Comment:
actually i think we can do this in a single query (and we should). this is because we're doing an `innerJoin` here on our largest table. 

i'm ok w/ accepting this now if it's difficult to do but concretely i think this entails: 
1. omitting the `where draftBlobId is not null` clause
2. return draft blob if exists, else published blob.

also, on further examination, will we ever have a case where we have `draftBlob` **but not** `publishedBlob`? since we likely just copy over, i don't really think this is a case right?

Reply:
might not need to do in a single query if it is hard to read, but suggest using a transaction instead.